metaflow/plugins/pypi/parsers.py

```diff
@@ -0,0 +1,260 @@
+import re
+from typing import Any, Dict, List, Optional
+
+REQ_SPLIT_LINE = re.compile(r"([^~<=>]*)([~<=>]+.*)?")
+
+# Allows things like:
+# pkg = <= version
+# pkg <= version
+# pkg = version
+# pkg = ==version or pkg = =version
+# In other words, the = is optional but possible
+YML_SPLIT_LINE = re.compile(r"(?:=\s)?(<=|>=|~=|==|<|>|=)")
+
+
+def req_parser(config_value: str) -> Dict[str, Any]:
```

Comment:
is it possible for extensions to define and override these parsers? that way this implementation can be decoupled with metaflow-netflixext

Reply:
There is no special mechanism right now. If this was something in core, the override we do for things like Parameter would work fine but here this is not. I don't think this will actually work right now (looking at it) because the plugin `pypi` doesn't exist internally (it's not loaded). I'm going to have to play around with it a bit. We could of course add a similar plugin category `PARSERS` (like everything else) and resolve it in the plugins. Bringing it back up here though will be a bit trickier.